specification/resources/resource-manager/Microsoft.Authorization/stable/2025-01-01/policyAssignments.json

```diff
@@ -0,0 +1,1076 @@
+          "enum": [
+            "NotSpecified",
+            "System",
+            "SystemHidden",
+            "Custom"
+          ],
+          "x-ms-enum": {
+            "name": "assignmentType",
+            "modelAsString": true
+          }
+        },
+        "instanceId": {
+          "type": "string",
+          "description": "The instance ID of the policy assignment. This ID only and always changes when the assignment is deleted and recreated.",
+          "readOnly": true
+        }
+      },
+      "description": "The policy assignment properties."
+    },
+    "NonComplianceMessage": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string",
+          "description": "A message that describes why a resource is non-compliant with the policy. This is shown in 'deny' error messages and on resource's non-compliant compliance results."
+        },
+        "policyDefinitionReferenceId": {
+          "type": "string",
+          "description": "The policy definition reference ID within a policy set definition the message is intended for. This is only applicable if the policy assignment assigns a policy set definition. If this is not provided the message applies to all policies assigned by this policy assignment."
+        }
+      },
+      "required": [
+        "message"
+      ],
+      "description": "A message that describes why a resource is non-compliant with the policy. This is shown in 'deny' error messages and on resource's non-compliant compliance results."
+    },
+    "ParameterValues": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/ParameterValuesValue"
+      },
+      "description": "The parameter values for the policy rule. The keys are the parameter names."
+    },
+    "ParameterValuesValue": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The value of the parameter.",
+          "type": "object"
+        }
+      },
+      "description": "The value of a parameter."
+    },
+    "PolicyAssignment": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/PolicyAssignmentProperties",
+          "description": "Properties for the policy assignment."
+        },
+        "id": {
+          "type": "string",
+          "description": "The ID of the policy assignment.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the policy assignment.",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of the policy assignment.",
+          "readOnly": true
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the policy assignment. Only required when utilizing managed identity."
```

Comment:
Regarding your suppressions. Proxy resource should not have a location. 

Arm will not route this accordingly and this location field will be confusing for users. How does this work currently ?

Reply:
PolicyAssignments sometimes need to have an identity (when it is meant to be able to make updates to other resources).  When an assignment has an identity, we require this location property to be set so that the frontdoor knows what MSI endpoint to talk to